src/js/node/stream.js

```diff
@@ -5279,7 +5279,7 @@ function createNativeStreamReadable(nativeType, Readable) {
 
       var ptr = this.#bunNativePtr;
       $debug("ptr @ NativeReadable._read", ptr, this.__id);
-      if (ptr === 0) {
+      if (ptr === 0 || ptr === -1) {
```

Comment:
maybe adding some comments for what -1 and 0 mean (Locked/Detached) so we can remember next time